watsononlinestore/watson_online_store.py

```diff
@@ -303,6 +305,19 @@ def get_discovery_response(self, input_text):
             query_options={'query': input_text, 'count': DISCOVERY_QUERY_COUNT}
         )
 
+        # Watson discovery assigns a confidence level to each result.
+        # Based on data mix, we can assign a minimum tolerance value in an
+        # attempt to filter out the "weakest" results.
+        if 'results' in discovery_response and self.discovery_score_filter:
```

Comment:
I think this would be nicer with self.disc..filter before the 'results' in (just flipping the sides of and)
if you are making a change anyway and if you agree.